thetis/callback.py

```diff
@@ -481,3 +481,43 @@ def __call__(self):
             field_vals.append(np.reshape(self._evaluate_field(field_name), (ndetectors, -1)))
 
         return np.hstack(field_vals)
+
+
+class FunctionalCallback(DiagnosticCallback):
```

Comment:
The name could be a bit more specific, functional sounds a bit too generic. It probably should indicate that this callback integrates (an existing) scalar callback over time.